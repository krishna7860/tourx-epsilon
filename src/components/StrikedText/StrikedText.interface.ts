interface StrikedTextProps {
  text: string;
}

export default StrikedTextProps;
