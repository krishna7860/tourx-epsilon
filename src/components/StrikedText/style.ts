import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  margin: 1.5rem 0rem;
`;

export const Span = styled.span`
  font-size: ${(props) => props.theme.fonts.textFade.size};
  color: ${(props) => props.theme.fonts.textFade.color};
  margin: 0rem 0.75rem;
`;

export const StyledHr = styled.hr`
  height: 1px;
  background: ${(props) => props.theme.fonts.textFade.color};
  border-radius: 10px;
  border: none;
  width: 100%;
`;
