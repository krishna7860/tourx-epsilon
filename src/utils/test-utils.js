/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { renderHook as rtlRenderHooks } from "@testing-library/react-hooks";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import * as themes from "../theme/schema.json";
import reducer from "../redux/rootReducer";
// import defaultStore from "../redux/defaultStore";

function render(ui, options = {}) {
  const {
    initialState = {},
    store = createStore(reducer, initialState),
    theme,
    ...renderOptions
  } = options;

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <ThemeProvider theme={themes.data.light}>{children}</ThemeProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// function renderHookWithRedux(callback, options = {}) {
//   const { initialState = defaultStore, store = createStore(reducer, initialState), ...others } = options

//   const returned = rtlRenderHooks(callback, {
//     ...others,
//     // eslint-disable-next-line react/display-name
//     wrapper: ({ children }) => <Provider store={store}>{children}</Provider>
//   })

//   return {
//     ...returned,
//     store
//   }
// }

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, rtlRender as renderHook };
