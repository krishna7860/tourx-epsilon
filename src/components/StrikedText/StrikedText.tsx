import React from "react";
import StrikedTextProps from "./StrikedText.interface";
import { Container, Span, StyledHr } from "./style";

const StrikedText = (props: StrikedTextProps): JSX.Element => {
  const { text } = props;
  return (
    <Container>
      <StyledHr />
      <Span>{text}</Span>
      <StyledHr />
    </Container>
  );
};

export default StrikedText;
