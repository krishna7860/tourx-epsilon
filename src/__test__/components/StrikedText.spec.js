/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { render } from "../../utils/test-utils";
import StrikedText from "../../components/StrikedText/StrikedText";

describe("components/StrikedText", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(<StrikedText text="sample text" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
