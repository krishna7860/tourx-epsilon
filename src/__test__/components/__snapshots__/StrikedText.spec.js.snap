// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StrikedText should be rendered properly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM ewqDOx"
  >
    <hr
      class="sc-dlnjwi cuisSO"
    />
    <span
      class="sc-gtsrHT ggiNGg"
    >
      sample text
    </span>
    <hr
      class="sc-dlnjwi cuisSO"
    />
  </div>
</DocumentFragment>
`;
